functional-tests.sh

```diff
@@ -678,6 +678,34 @@ function test_cat_object_with_sse_error()
     log_success "$start_time" "${FUNCNAME[0]}"
 }
 
+# Test mc cp command with preserve file system attributes
+function test_copy_object_preserve_filesystem_attr()
+{
+    show "${FUNCNAME[0]}"
+    start_time=$(get_time)
+    object_name="mc-test-object-$RANDOM"
+    atime="$(stat --format=%X ${FILE_1_MB})"
+    ctime="$(stat --format=%Y ${FILE_1_MB})"
+    gid="$(stat --format=%g ${FILE_1_MB})"
+    gname="$(stat --format=%G ${FILE_1_MB})"
+    md5_sum="$(get_md5sum "$FILE_1_MB")"
+    if [ $? -ne 0 ]; then
+        echo "unable to get md5sum of $FILE_1_MB"
+        exit 1
+    fi
+    mode="$(stat --format=%a ${FILE_1_MB})"
+    mtime="$(stat --format=%Z ${FILE_1_MB})"
+    userid="$(stat --format=%u ${FILE_1_MB})"
+    uname="$(stat --format=%U ${FILE_1_MB})"
+    FILE_ATTRIBUTE="atime:${atime}/ctime:${ctime}/gid:${gid}/gname:${gname}/md5:${md5_sum}/mode:${mode}/mtime:${mtime}/uid:${userid}/uname:${uname}"
```

Comment:
This test is the wrong @sinhaashish using Unix stat command will fail because it is not converting the octal representation properly instead it will show you instead enhance `mc stat --json` output to print the `mode` of the file instead @sinhaashish 

Or enhance the `mc stat` itself to print in the `FILE_ATTRIBUTE` fashion such that comparison becomes easier.